oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -200,9 +217,23 @@ static Map<String, List<String>> addQuotaProjectIdToRequestMetadata(
     return Collections.unmodifiableMap(newRequestMetadata);
   }
 
+  @Override
+  protected Map<String, List<String>> getAdditionalHeaders() {
+    Map<String, List<String>> headers = super.getAdditionalHeaders();
+    if (quotaProjectId != null) {
```

Comment:
Should this use `this.getQuotaProjectId()`. What if a child class overrides how it stores/returns the quota project id?